### ISSUE TITLE
Bug: Fresh install with pnpm installs multiple versions of typescript-estree

### ISSUE THREAD

**Issue by Mitsunee** [typescript-eslint/typescript-eslint#9750]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

typescript-estree

### Repro Code

I am currently working on upgrading multiple projects (as I do every month) and have just upgraded all typescript-eslint packages to 8.0.1. I have not upgraded the typescript package. 

This upgrade resulted in both `@typescript-eslint/typescript-estree@7.2.0` and `@typescript-eslint/typescript-estree@8.0.1` being installed simultanously, which remains the case after `pnpm dedupe`. I have also tried deleting `node_modules`, `pnpm-lock.yaml` and running `pnpm store prune` before reinstalling and deduping.

This results in an older version of `typescript-estree` being used, giving a warning about an unsupported version of TypeScript being used, despite the fact that support for TS 5.5 has landed some versions ago and has previously worked in the very same project.

I also tested the same project with yarn classic and did not run into this issue.

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/<rule-name>": ["error", ...<options>],
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

only the correct version is installed, mismatched versions are ignored

### Actual Result

older version of typescript-estree is installed and used

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `8.0.1` |
| `@typescript-eslint/parser`            | `8.0.1` |
| `@typescript-eslint/typescript-estree` | `8.0.1` (as well as incorrectly installed `7.2.0`) |
| `TypeScript`                           | `5.5.4` |
| `ESLint`                               | `8.57.0` |
| `node`                                 | `20.14.0` |

**Comment by JoshuaKGoldberg** [comment]
Hey glad to see you're trying out v8 & updating regularly! But, given that this issue is about versioning, we'd need to see a [reproduction that reproduces the issue](https://antfu.me/posts/why-reproductions-are-required#reproducible-projects-or-playgrounds).

I don't reproduce this in a fresh directory with `pnpm i typescript-eslint`, nor after adding `pnpm i @typescript-eslint/eslint-plugin @typescript-eslint/parser`.

**Comment by Mitsunee** [comment]
I am unsure as to how to provide a reproduction for pnpm's cli, fwiw global packages are the first thing I updated today.

**Comment by bradzacher** [comment]
Our `@typescript-eslint/utils` package depends on `@typescript-eslint/typescript-estree`.

It's common for 3rd party plugins to depend on `/utils` - which means that even if you depend on v8 of our tooling, a plugin you've installed could depend on v7 of `/utils` which means v7 of `/typescript-estree`.

So I'd say this is completely expected - esp given many of the 3rd parties haven't yet upgraded to our v8 release yet cos it's only been a week.

**Comment by Mitsunee** [comment]
found the user error, one of the plugins in those projects was also installing all of typescript-eslint 7.2.0. I'm guessing yarn just doesn't care and let's me decide to use a newer version, so this is actually an issue for pnpm, my bad.

**Comment by JoshuaKGoldberg** [comment]
Great!

> how to provide a reproduction for pnpm's cli

In case it's useful in the future, I personally a branch on a personal reproduction repository: https://github.com/JoshuaKGoldberg/repros. You could alternately use a service like Stackblitz.
